### DISCUSSION TITLE
What is the <csr-id/> tag for?

### DISCUSSION THREAD

**Discussion by juliusl** [GitoxideLabs/cargo-smart-release#24] [Q&A] [answered]
In Azure devops for some reason, the markdown renderer will actually show the tag. I was wondering what these tags were used for?

**Comment by Byron** [comment, marked as answer]
Every message that it presents is based on a commit, and the commit-hash essentially the `csr-id` - don't ask me know what `csr` was supposed to stand for, but probably something with `content`.

These are used to allow the editor to remove certain messages, and as long as the csr-id is left in the document, it will not re-appear - after all, the generated markdown is fully editable.

I hope that helps - it's indeed unfortunate that these are rendered by Azure Devops.

**Reply by juliusl** [reply]
Thanks for the quick reply. Definitely sounds like something I want to leave in.

(I was guessing csr stood for cargo-smart-release)

**Reply by Byron** [reply]
> (I was guessing csr stood for cargo-smart-release)

Hehe, yes, that's probably it. Kind of obvious, yet I couldn't imagine it has nothing to do with `content…` :D.
